walk-in-interview/src/main/java/com/google/account/FirebaseAuthSessionCookieUtil.java

```diff
@@ -0,0 +1,125 @@
+            SessionCookieOptions options = SessionCookieOptions.builder().setExpiresIn(expiresIn).build();
+
+            // Creates the session cookie. This will also verify the ID token in the process.
+            // The session cookie will have the same claims as the ID token.
+            String sessionCookie = FirebaseAuth.getInstance().createSessionCookieAsync(idToken, options).get();
+
+            // Sets cookie policy parameters as required
+            NewCookie cookie = new NewCookie(/* name= */ "session", sessionCookie);
+            return Response.ok().cookie(cookie).build();
+        } catch (Exception e) {
+            return Response.status(Response.Status.UNAUTHORIZED).entity("Failed to create a session cookie").build();
+        }
+    }
+
+    /**
+     * Verifies session cookies.
+     *
+     * @param cookie Session cookie.
+     * @return Response with uid of the current user account.
+     */
+    @POST
+    @Path("/profile")
+    public Response verifySessionCookie(@CookieParam("session") Cookie cookie) {
+        String sessionCookie = cookie.getValue();
+        try {
+            // Verify the session cookie. In this case an additional check is added to detect
+            // if the user's Firebase session was revoked, user deleted/disabled, etc.
+            final boolean checkRevoked = true;
+            FirebaseToken decodedToken = FirebaseAuth.getInstance().verifySessionCookie(sessionCookie, checkRevoked);
+            return getUid(decodedToken);
+        } catch (FirebaseAuthException e) {
+            // Session cookie is unavailable, invalid or revoked. Force user to login.
+            return Response.temporaryRedirect(URI.create(LOG_IN_PAGE_PATH)).build();
+        }
+    }
+
+    /**
+     * Clears the session cookie when the user signs out at the client side.
+     *
+     * @param cookie Session cookie
+     * @return Response to direct to log in page.
+     */
+    @POST
+    @Path("/business-log-out")
+    public Response clearSessionCookie(@CookieParam("session") Cookie cookie) {
+        final int maxAge = 0;
+        NewCookie newCookie = new NewCookie(cookie, /* comment= */ null, maxAge, /* secure= */ true);
+        return Response.temporaryRedirect(URI.create(LOG_IN_PAGE_PATH)).cookie(newCookie).build();
+    }
+
+    /** Initializes the admin SDK. */
+    private void initAdminSDK() throws IOException {
+        FirebaseOptions options = new FirebaseOptions.Builder()
+                .setCredentials(GoogleCredentials.getApplicationDefault())
+                .setDatabaseUrl("https://com-walk-in-interview.firebaseio.com/")
+                .setProjectId("com-walk-in-interview")
+                .build();
+
+        FirebaseApp.initializeApp(options);
+    }
+
+    /** Gets the id token from the log in request. */
+    private String getIDToken(HttpServletRequest request) throws IllegalArgumentException {
+        String idToken = request.getParameter(ID_TOKEN_PARAM).trim();
```

Comment:
maybe you can use something like: 
`        String jobIdStr = ServletUtils.getStringParameter(request, JOB_ID_FIELD, /* defaultValue= */ "");
`